### PR TITLE
CI: speed up e2e tests

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong <loong.dai@intel.com>

# Description

Build one time to reduce.

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
Compared with https://github.com/dapr/dapr/pull/4088, the time of the e2e test reduces.

File: docker/docker.mk [hunk 1]
```diff
@@ -121,6 +121,14 @@ else
 	$(DOCKER) buildx build --build-arg PKG_FILES=sentry --platform $(DOCKER_IMAGE_PLATFORM) -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(BIN_PATH) -t $(DAPR_SENTRY_DOCKER_IMAGE_TAG)-$(TARGET_OS)-$(TARGET_ARCH) --push
 endif
 
+# push docker image for e2e tests
+docker-push-e2e:
```
Comment (wcs1only) [inline docker/docker.mk:125]: Why do we need a separate target for this? Isn't docker-push above sufficient?

File: .github/workflows/kind-e2e.yaml [hunk 2]
```diff
@@ -37,8 +37,39 @@ jobs:
...
+    - name: Set up Go ${{ env.GOVER }}
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GOVER }}
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@v2
+      with:
+        path: ./src/github.com/dapr/dapr
+    # setup image cache
+    - uses: satackey/action-docker-layer-caching@v0.0.11
```
Comment (wcs1only) [inline .github/workflows/kind-e2e.yaml:60]: We have attempted this in the past. Trouble was the Windows images being extremely large ended up paving the cache and breaking this optimization. But, let's give it a try and see what happens.

File: docker/docker.mk (hunk 1 above)
Comment (daixiang0) [reply docker/docker.mk:125]: `docker-push` depends on `docker-build` :(

File: .github/workflows/kind-e2e.yaml (hunk 2 above)
Comment (daixiang0) [reply .github/workflows/kind-e2e.yaml:60]: It is for kind e2e test which only run at Linux I think.

**Comment by daixiang0** [comment]
@wcs1only could we make it land?

File: docker/docker.mk (hunk 1 above)
Comment (artursouza) [reply docker/docker.mk:125]: I would rather create a special env variable that will skip the build and keep the docker-push target above only. There is more logic to support multi arch, for example, that this one does not replicate.

**Review by artursouza — CHANGES_REQUESTED** [review]

**Pull request closed without merging**